### PR TITLE
fix: URL encoding [HEAD-211][HEAD-212]

### PR DESCRIPTION
**Pull request by asaf92**
### Description

Fix URL Encoding in `limitToFiles` for Code API calls

### CODE REVIEW

File: infrastructure/code/code.go [hunk 1]
```diff
@@ -488,14 +479,29 @@ func (sc *Scanner) createBundle(
 		if !(len(fileContent) > 0 && len(fileContent) <= maxFileSize) {
 			continue
 		}
-		file := getFileFrom(filePath, fileContent)
-		bundleFiles[filePath] = file
-		fileHashes[filePath] = file.Hash
+		bundleFile := getFileFrom(filePath, fileContent)
+		bundleFiles[filePath] = bundleFile
+		fileHashes[filePath] = bundleFile.Hash
+		fileName := filepath.Base(filePath)
```
Comment (bastiandoetsch) [inline infrastructure/code/code.go:485]: this is most likely not correct - it should be the path from folder root - e.g. if you have several `index.js` it wouldn't work, if I understand correctly.

File: infrastructure/code/bundle.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ type Bundle struct {
 	SnykCode      SnykCodeClient
 	BundleHash    string
 	UploadBatches []*UploadBatch
+	Files         map[string]BundleFile
```
Comment (bastiandoetsch) [inline infrastructure/code/bundle.go:39]: why do we need to have a copy of the files here?

File: infrastructure/code/code_test.go [hunk 3]
```diff
@@ -75,11 +75,16 @@ func TestCreateBundle(t *testing.T) {
...
+				"testRequestId",
+				dir,
+				[]string{file},
+				map[string]bool{})
 			if err != nil {
 				t.Fatal(err)
 			}
-			assert.Len(t, missingFiles, 1, "bundle should have 1 missing files")
+			bundleFiles := bundle.Files
+			assert.Len(t, bundleFiles, 1, "bundle should have 1 bundle files")
```
Comment (bastiandoetsch) [inline infrastructure/code/code_test.go:87]: missing files != bundle files

File: infrastructure/code/code_test.go [hunk 4]
```diff
@@ -92,11 +97,16 @@ func TestCreateBundle(t *testing.T) {
...
+				"testRequestId",
+				dir,
+				[]string{file},
+				map[string]bool{})
 			if err != nil {
 				t.Fatal(err)
 			}
-			assert.Len(t, missingFiles, 0, "bundle should not have missing files")
+			bundleFiles := bundle.Files
+			assert.Len(t, bundleFiles, 0, "bundle should not have bundle files")
```
Comment (bastiandoetsch) [inline infrastructure/code/code_test.go:109]: same here - missing files is what it is about - not bundle files

File: infrastructure/code/code_test.go (hunk 3 above)
Comment (asaf92) [reply infrastructure/code/code_test.go:87]: It's supposed to be "bundle files". The missing files are coming from a component that is not under test here (the snyk code client is mocked). It always tested for bundle files, nothing was changed logic-wise

**Pull request merged**